src/data/FieldRegistry.tsx

```diff
@@ -4,14 +4,27 @@ import { IFieldOptionEditorComponent } from './IFieldOptionEditor'
 import { IFieldInputComponent } from './IFieldInput'
 import { IFieldDisplayComponent } from './IFieldDisplay'
 
+export interface IFieldDef {
+    type: string;
+    displayName: string;
+    field: IField;
+    builder: IFieldBuilderComponent;
+    input: IFieldInputComponent;
+    display: IFieldDisplayComponent;
+    editor?: IFieldOptionEditorComponent;
+}
+
 // FieldRegistry maps field types to the class responsible for rendering the field and the class responsible for editing the field.
-export interface FieldRegistry {
-    [type: string]: {
-        field: IField,
-        displayName: string,
-        builder: IFieldBuilderComponent,
-        input: IFieldInputComponent,
-        display: IFieldDisplayComponent,
-        editor?: IFieldOptionEditorComponent,
-    };
+export class FieldRegistry {
+    // NOTE(gaolw): 'any' here is to make typescript happy because the 'register' property is not IFieldDef.
+    // More details: please refer to https://www.typescriptlang.org/docs/handbook/interfaces.html
+    [type: string]: IFieldDef | any;
```

Comment:
Since we're creating a class for the registry, I suggest creating a getter method for retrieving components and making the map private